src/lib.rs

```diff
@@ -30,3 +35,17 @@ pub fn format_duration(duration: &chrono::Duration) -> String {
         days, hours, minutes, seconds, milliseconds
     )
 }
+
+pub trait BytesMutReader {
+    fn read_string(&mut self) -> Result<String, Error>;
+}
+
+impl BytesMutReader for Cursor<&BytesMut> {
+    fn read_string(&mut self) -> Result<String, Error> {
```

Comment:
What if the query contains a null byte? I think we need to be careful here and read the exact amount Postrgres is telling us is in the message.